obal/data/roles/build_package/tasks/copr.yml

```diff
@@ -0,0 +1,65 @@
+---
+- name: 'Set test build'
+  set_fact:
+      build_package_tito_args: "{{ build_package_tito_args }} --test"
+  when: build_package_test|default(false)
+  tags:
+    - always
+
+- block:
+    - name: 'Set tito_releasers for copr release'
+      set_fact:
+        build_package_tito_releasers: "{{ releasers | join(' ') }}"
+      tags:
+        - always
+
+    - include_tasks: tito_release.yml
+      tags:
+        - always
+  when: not build_package_scratch
+
+- block:
+    - name: 'Create .tmp build directory'
+      file:
+        state: directory
+        path: "{{ inventory_dir }}/.tmp"
+      run_once: true
+      tags:
+        - always
+
+    - set_fact:
+        copr_repo_name: "{{ copr_user }}/{{ build_package_scratch_repo}}-{{ 999999999 | random | to_uuid }}"
+      run_once: true
+      when: not copr_repo_name
+      tags:
+        - always
+
+    - name: 'Write copr repo name to vars file'
+      copy:
+        content: "{{ 'copr_repo: ' + copr_repo_name | to_yaml }}"
+        dest: "{{ inventory_dir }}/.tmp/copr_repo"
+      run_once: true
+
+    - include_role:
+        name: copr_repo
+      run_once: true
+      tags:
+        - always
+
+    - name: 'Build SRPM'
+      command: 'tito build --srpm --scl={{ scl }}'
+      args:
+        chdir: "{{ inventory_dir }}/packages/{{ inventory_hostname }}"
+      register: srpm_build
+      tags:
+        - always
+
+    - name: 'Run build'
+      command: "copr-cli build {{ copr_repo_name }} {{ srpm_build.stdout.split('Wrote: ')[1] }}"
```

Comment:
@evgeni I need a little testing help. This line here expects the previous task to have created an SRPM and stored that information in a variable that this extracts. The test framework with mockbin currently does not have any mocked output which results in `srpm_build.stdout` being empty string and then this fails during testing.

Reply:
Well, then let's mock `tito` a bit more, shall we?

```diff
diff --git a/tests/fixtures/mockbin/mockbin b/tests/fixtures/mockbin/mockbin
index a7223ac..22708ca 100755
--- a/tests/fixtures/mockbin/mockbin
+++ b/tests/fixtures/mockbin/mockbin
@@ -1,10 +1,29 @@
-#!/usr/bin/env python2
+#!/usr/bin/env python
+
+from __future__ import print_function
 
 import os
 import sys
+import argparse
 
 mockbin_log = os.environ.get('MOCKBIN_LOG', None)
 
 if mockbin_log:
     with open(mockbin_log, 'a') as logfile:
         logfile.write("%s\n" % sys.argv)
+
+prog = os.path.basename(sys.argv[0])
+
+if prog == 'tito':
+    parser = argparse.ArgumentParser()
+    subparsers = parser.add_subparsers()
+    parser_release = subparsers.add_parser('release')
+    parser_release.add_argument('dest')
+    parser_release.add_argument('--scratch', action='store_true')
+    parser_release.add_argument('-y', action='store_true')
+    parser_build = subparsers.add_parser('build')
+    parser_build.add_argument('--srpm', action='store_const', dest='output', const='Wrote: hello.src.rpm')
+    parser_build.add_argument('--scl', action='store')
+    args = parser.parse_args()
+    if 'output' in args and args.output:
+        print(args.output)
diff --git a/tests/test_functional.py b/tests/test_functional.py
index 177e86d..19a9667 100644
--- a/tests/test_functional.py
+++ b/tests/test_functional.py
@@ -208,7 +208,7 @@ def test_obal_scratch_copr_hello():
         "['{bin}/copr-cli', 'create', 'copr-repo-scratch', '--chroot', 'epel-7-x86_64', '--description', 'Scratch Builds', '--unlisted-on-hp', 'on', '--repo', 'http://mirror.centos.org/centos/7/sclo/x86_64/rh/']",  # noqa: E501
         "['{bin}/copr-cli', 'edit-chroot', 'copr-repo-scratch/epel-7-x86_64', '--packages', 'scl-utils-build rh-ruby24-build']",  # noqa: E501
         "['{bin}/tito', 'build', '--srpm', '--scl=copr-scl']",
-        "['{bin}/copr-cli', 'build', 'copr-repo-scratch']"
+        "['{bin}/copr-cli', 'build', 'copr-repo-scratch', 'hello.src.rpm']"
     ]
     assert_mockbin_log(expected_log)
 
```